### ISSUE TITLE
Package is broken

### ISSUE THREAD

**Issue by ProgVal** [ProjetPP/PPP-QuestionParsing-Grammatical#100]
I think the data is not shipped with the packages:

```
: Traceback (most recent call last):
  File "/home/ppp/.local/lib/python3.4/site-packages/ppp_core/router.py", line 112, in _get_python
    obj = self._get_python_class(module.url)(request)
  File "/home/ppp/.local/lib/python3.4/site-packages/ppp_core/router.py", line 103, in _get_python_class
    cls = importlib.import_module(module_path)
  File "/usr/lib/python3.4/importlib/__init__.py", line 109, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 2254, in _gcd_import
  File "<frozen importlib._bootstrap>", line 2237, in _find_and_load
  File "<frozen importlib._bootstrap>", line 2212, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 321, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 2254, in _gcd_import
  File "<frozen importlib._bootstrap>", line 2237, in _find_and_load
  File "<frozen importlib._bootstrap>", line 2226, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1200, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1129, in _exec
  File "<frozen importlib._bootstrap>", line 1471, in exec_module
  File "<frozen importlib._bootstrap>", line 321, in _call_with_frames_removed
  File "/home/ppp/.local/lib/python3.4/site-packages/ppp_questionparsing_grammatical/__init__.py", line 4, in <module>
    from .preprocessingMerge import mergeNamedEntityTagChildParent, mergeNamedEntityTagSisterBrother, mergeNamedEntityTag
  File "/home/ppp/.local/lib/python3.4/site-packages/ppp_questionparsing_grammatical/preprocessingMerge.py", line 11, in <module>
    nManual.load(os.path.join(os.path.dirname(__file__), 'data/nounificationManual.pickle'))
  File "/home/ppp/.local/lib/python3.4/site-packages/ppp_questionparsing_grammatical/nounDB.py", line 18, in load
    f = open(file_name, 'rb')
FileNotFoundError: [Errno 2] No such file or directory: '/home/ppp/.local/lib/python3.4/site-packages/ppp_questionparsing_grammatical/data/nounificationManual.pickl
```

You might be interested in reading this: https://pythonhosted.org/setuptools/setuptools.html#including-data-files

**Comment by yhamoudi** [comment]
how did you get this error (how can i reproduce it) ?

**Comment by ProgVal** [comment]
```
pip3 uninstall ppp_questionparsing_grammatical
cd ~/etudes/ens/ppp/modules/grammatical
make localinstall
cd /tmp/
PPP_QUESTIONPARSING_GRAMMATICAL_CONFIG=/home/progval/etudes/ens/ppp/modules/grammatical/example_config.json python3 -c "import ppp_questionparsing_grammatical"
```

**Comment by yhamoudi** [comment]
i obtain an error message: 
```

**Comment by Ezibenroc** [comment]
You need to set the environment variable. You can do as follow:
```
PPP_QUESTIONPARSING_GRAMMATICAL_CONFIG=/path/to/your/example_config.json python3 -c "import ppp_questionparsing_grammatical"
```

I think this fix the bug: https://github.com/ProjetPP/PPP-QuestionParsing-Grammatical/commit/cd4c1374d54a650eead3b941e46d33679c4ee68d

**Comment by ProgVal** [comment]
Now I can't even install the package:

```
progval@ganymede:~/etudes/ens/ppp/modules/grammatical(master)$ make localinstall 
python3 setup.py install --user
running install
running bdist_egg
running egg_info
writing top-level names to ppp_questionparsing_grammatical.egg-info/top_level.txt
writing dependency_links to ppp_questionparsing_grammatical.egg-info/dependency_links.txt
writing ppp_questionparsing_grammatical.egg-info/PKG-INFO
writing requirements to ppp_questionparsing_grammatical.egg-info/requires.txt
reading manifest file 'ppp_questionparsing_grammatical.egg-info/SOURCES.txt'
writing manifest file 'ppp_questionparsing_grammatical.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
running build_py
creating build
creating build/lib
creating build/lib/ppp_questionparsing_grammatical
copying ppp_questionparsing_grammatical/dependencyAnalysis.py -> build/lib/ppp_questionparsing_grammatical
copying ppp_questionparsing_grammatical/config.py -> build/lib/ppp_questionparsing_grammatical
copying ppp_questionparsing_grammatical/normalization.py -> build/lib/ppp_questionparsing_grammatical
copying ppp_questionparsing_grammatical/questionWordProcessing.py -> build/lib/ppp_questionparsing_grammatical
copying ppp_questionparsing_grammatical/preprocessing.py -> build/lib/ppp_questionparsing_grammatical
copying ppp_questionparsing_grammatical/requesthandler.py -> build/lib/ppp_questionparsing_grammatical
copying ppp_questionparsing_grammatical/nounDB.py -> build/lib/ppp_questionparsing_grammatical
copying ppp_questionparsing_grammatical/preprocessingMerge.py -> build/lib/ppp_questionparsing_grammatical
copying ppp_questionparsing_grammatical/__init__.py -> build/lib/ppp_questionparsing_grammatical
creating build/lib/ppp_questionparsing_grammatical/data
copying ppp_questionparsing_grammatical/data/exceptions.py -> build/lib/ppp_questionparsing_grammatical/data
copying ppp_questionparsing_grammatical/data/superlative.py -> build/lib/ppp_questionparsing_grammatical/data
copying ppp_questionparsing_grammatical/data/conjunction.py -> build/lib/ppp_questionparsing_grammatical/data
copying ppp_questionparsing_grammatical/data/questionWord.py -> build/lib/ppp_questionparsing_grammatical/data
copying ppp_questionparsing_grammatical/data/__init__.py -> build/lib/ppp_questionparsing_grammatical/data
copying ppp_questionparsing_grammatical/data/nounificationManual.pickle -> build/lib/ppp_questionparsing_grammatical/data
error: can't copy 'ppp_questionparsing_grammatical/data/__pycache__': doesn't exist or not a regular file
Makefile:10: recipe for target 'localinstall' failed
make: *** [localinstall] Error 1
```

**Comment by Ezibenroc** [comment]
Fix: https://github.com/ProjetPP/PPP-QuestionParsing-Grammatical/commit/75a6b6dec41f31505e9041d2f0d8dfc8bdbd4f12

**Comment by ProgVal** [comment]
Still getting the error from the beginning.
Did you read this? https://pythonhosted.org/setuptools/setuptools.html#accessing-data-files-at-runtime

**Comment by ProgVal** [comment]
Fixed in 17cd634.

**Comment by ProgVal** [comment]
Hmm, I thought it was a different bug, but it seems highly related:

```
Traceback (most recent call last):
  File "/usr/lib/python3.4/unittest/loader.py", line 312, in _find_tests
    module = self._get_module_from_name(name)
  File "/usr/lib/python3.4/unittest/loader.py", line 290, in _get_module_from_name
    __import__(name)
  File "/home/progval/etudes/ens/ppp/modules/grammatical/tests/test_deep.py", line 6, in <module>
    from ppp_questionparsing_grammatical import app
  File "/home/progval/etudes/ens/ppp/modules/grammatical/ppp_questionparsing_grammatical/__init__.py", line 4, in <module>
    from .preprocessingMerge import mergeNamedEntityTagChildParent, mergeNamedEntityTagSisterBrother, mergeNamedEntityTag
  File "/home/progval/etudes/ens/ppp/modules/grammatical/ppp_questionparsing_grammatical/preprocessingMerge.py", line 12, in <module>
    nManual.load(resource_string('ppp_questionparsing_grammatical', 'data/nounificationManual.pickle'))
  File "/home/progval/etudes/ens/ppp/modules/grammatical/ppp_questionparsing_grammatical/nounDB.py", line 18, in load
    f = open(file_name, 'rb')
TypeError: embedded NUL character
```

**Comment by Ezibenroc** [comment]
I don't have any error, either on my PC or on angel...

**Comment by ProgVal** [comment]
I don't know how you tried to reproduce it, but the code I committed could not run at all, whatever the configuration or the Python version…

Anyway, I fixed it in ebdb101.
